schema.json

```diff
@@ -1,12 +1,13 @@
 {
     "$schema": "http://json-schema.org/draft-08/schema#",
     "$id": "http://json-schema.org/draft-08/schema#",
+
     "title": "Core schema meta-schema",
     "$defs": {
         "schemaArray": {
             "type": "array",
             "minItems": 1,
-            "items": { "$ref": "#" }
+            "items": { "$recursiveRef": "#" }
```

Comment:
schema.json does not need `"$recursiveRoot": true`, as it is not extending anything.  So using the literal URI reference of `#` in `"$recursiveRef": "#"` produces the correct behavior when using schema.json directly.

When used from hyper-schema.json (with either it, links.json, or hyper-operations.json as the entry schema document), a `"$recursiveRoot"` will be set, so it is important that these references are `"$recursiveRef"` rather than `"$ref"`.